### DISCUSSION TITLE
Features / ideas about bash support and server installation process.

### DISCUSSION THREAD

**Discussion by mhclaudiu** [pufferpanel/pufferpanel#1230] [Ideas]
Hey people, first of all great work with this panel! Combining go with javascript is awesome!
I would like to propose 2 features/ideas:

1. Add the possibility so we can have full bash commands with bash variables supported in this block:
`"install": [
    {
      "commands": [
		"echo $(pwd) && chmod +x file"`

2. After we initiated the installation process for a server, and the installation ended successfully, it would be nice a feature so we can't click again by mistake on the Install button.

**Comment by nepcore** [comment]
1. `bash -c 'echo $(pwd)'` ;)
2. That's intentionally not a thing  
minecraft servers set to use the latest version and steam games use that to do updates  
minecraft servers, after changing the version setting, need to run that again to actually get that different version of minecraft  
and the list is only going to grow in the future

**Reply by mhclaudiu** [reply]
Thank you for your response!
So we can actually use full bash commands, great!
Hmm, so there are servers for which the Install button also checks and installs updates, okay. Well In this case i will modify the template and add a separate script in the Install section which will prevent and ask me twice if I really want to do this, if server was previously successfully installed.
